zetaclient/zetacore/background_threads.go

```diff
@@ -0,0 +1,162 @@
+package zetacore
+
+import (
+	"context"
+	"time"
+
+	"cosmossdk.io/errors"
+
+	"github.com/zeta-chain/zetacore/pkg/retry"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+	zctx "github.com/zeta-chain/zetacore/zetaclient/context"
+)
+
+// HandleTSSUpdate is a background thread that listens for TSS updates; it returns when the TSS address is updated
+func (c *Client) HandleTSSUpdate(ctx context.Context) error {
```

Comment:
Name is not reflective, it doesn't handle a TSS update, just wait for a TSS update
- ListenTSSUpdate
- WaitForTSSUpdate

Reply:
The reason for adding the `handle` keyword is the fact that when the Update happens, this routine triggers a shutdown for the main thread, using the `cancel()` function of the parent context.
Happy to change it if you still feel otherwise , resolving the other similar comments for now